### DISCUSSION TITLE
Validate post sub item in AdonisJs

### DISCUSSION THREAD

**Discussion by marcelodelta** [adonisjs/core#3507] [Help] [answered]
Validate post sub item in AdonisJs.

I can't validate a sub item of the post. I believe it is not found.

I need to validate "codigo" that is inside "nfe"

Request

```
{
  nfe: {
    codigo: '56456',
    motorista: 'NOME',
    contato: 'CONTATO',
    casa: true
  },
  fornecedor: 'ALIMENTOS',
  lab: 'sim',
}
```

Controller

`await request.validate(NfeValidator)`

Validator

```
public schema = schema.create({
    'nfe.codigo': schema.string({trim: true}, [rules.unique({table: 'nfes', column: 'codigo'})])
  })
```

**Comment by arthur-er** [comment, marked as answer]
You can use the [`schema.object`](https://docs.adonisjs.com/reference/validator/schema/object) to validate objects, the shape would look something like this
```typescript
nfe: schema.object().members({
    codigo: schema.string({ trim: true }, [rules.unique({ table: 'nfes', column: 'codigo' })]),
})
```
